File: payment-service/src/main/java/com/study/payment/domain/repository/PaymentRepository.java
package com.study.payment.domain.repository;

import com.study.payment.domain.entity.Payment;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface PaymentRepository extends JpaRepository<Payment, Long> {
    Optional<Payment> findByBookingId(Long bookingId);
    List<Payment> findByUserId(Long userId);
}

File: payment-service/src/main/kotlin/com/study/payment/domain/repository/PaymentRepository.kt
package com.study.payment.domain.repository

import com.study.payment.domain.entity.Payment
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface PaymentRepository : JpaRepository<Payment, Long> {
    fun findByBookingId(bookingId: Long): Optional<Payment>
    fun findByUserId(userId: Long): List<Payment>
}
